convert2rhel/toolopts.py

```diff
@@ -464,5 +496,43 @@ def options_from_config_files(cfg_path=None):
     return supported_opts
 
 
+def _parse_subscription_manager_serverurl(serverurl):
+    """Parse a url string in a manner mostly compatible with subscription-manager --serverurl."""
+    # This is an adaptation of what subscription-manager's cli enforces:
+    # https://github.com/candlepin/subscription-manager/blob/main/src/rhsm/utils.py#L112
+
+    # Don't modify http://<something> and https://<something> as they are fine
+    if not re.match("https?://[^/]+", serverurl):
+        # Anthing that looks like a malformed scheme is immediately discarded
+        if re.match("^[^:]+:/.+", serverurl):
+            raise ValueError("serverurl looks unparsable. Make sure it starts with http://HOST or https://HOST")
```

Comment:
Does anyone know a way to rephrase this? The sentence starting with a lowercase seems strange. Maybe:

```suggestion
            raise ValueError("Problem while parsing serverurl. Make sure it starts with http://HOST or https://HOST")
```

Reply:
```suggestion
            raise ValueError("Unable to parse --serverurl.  Make sure it starts with http://HOST or https://HOST")
```